packages/baseui-codemods/README.md

```diff
@@ -0,0 +1,159 @@
+
+There are a few ways to test your codemod while in development. Consider both methods requirements for adding a new codemod.
+
+The first method is to unit test your logic. We use [inline snapshots](https://jestjs.io/docs/en/snapshot-testing#inline-snapshots) to verify that codemod logic runs as expected. Reference other test files to get the general pattern we expect in these tests.
+
+The second method is to run your codemod on actual source code.
+
+Add your codemod to the `MODS` object in `src/cli.js`.
+
+```js
+// src/cli.js
+
+import {redToBlue, greenToPink} from './colorUpdate.js';
+
+const MODS = {
+  // ...
+  colorUpdate: [
+    step('replace "red" with "blue"', redToBlue),
+    step('replace "green" with "pink"', greenToPink),
+  ],
+};
+
+// more CLI logic below
+```
+
+This makes the `colorUpdate` codemod available on the command line.
+
+Run `yarn build` to compile everything into `dist` and now you can run `dist/cli.js` as you would any Node script.
+
+```shell
+$ node dist/cli.js --dir=~/code/some-project/src --mod=colorUpdate
+```
+
+You could also use `yarn link` and open a shell in the project directory. This is a bit closer to what an end user will experience.
+
+```shell
+$ baseui-codemods --dir=src --mod=colorUpdate
+```
+
+After running your codemod, you can use a diff tool to verify that the codebase has changed as expected. It is crucial to run every codemod against a number of projects to ensure it functions correctly. There are always edge-cases!
+
+#### References
+
+For some more info on developing CLI tools in node, check out [this article](https://medium.com/netscape/a-guide-to-create-a-nodejs-command-line-package-c2166ad0452e#c2d2). This project follows a lot of the recommendations in the article.
+
+A very useful tool for iterating on AST related logic is https://astexplorer.net/. Set the compiler to `Babylon7` and enable `Transform`, which will open two new panes. The bottom left is a babel plugin you can fill in and the bottom right is the output of that plugin on the source in the top left pane. The top right pane is for exploring the source code's AST, which will prove very useful if doing any sort of AST traversal in your codemod.
+
+For more information on AST traversal, you will want to become familiar with [how Babel plugins work](https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md), specifically **Traversal** via the **Visitor** pattern.
+
+#### Publishing
+
+Make sure you are logged in to the npm account which is part of the `@uber-web-ui` org.
```

Comment:
Should replace it with `reach out to us for publishing`?

Reply:
yeah i'll remove this- going to leave in the version policy just for clarification though